Assets/Scripts/Projects/ProjectData.cs

```diff
@@ -3,85 +3,92 @@
 using System.Runtime.InteropServices;
 using ICSharpCode.SharpZipLib.Core;
 using ICSharpCode.SharpZipLib.Zip;
-using Netherlands3D.Twin.Functionalities;
 using Netherlands3D.Twin.Layers;
-using Netherlands3D.Twin.UI.LayerInspector;
 #if UNITY_EDITOR
 using UnityEditor;
 #endif
 using UnityEngine;
 using UnityEngine.Events;
 using Newtonsoft.Json;
-using Application = UnityEngine.Application;
+using UnityEngine.Assertions;
 
 namespace Netherlands3D.Twin.Projects
 {
     [CreateAssetMenu(menuName = "Netherlands3D/Twin/Project", fileName = "Project", order = 0)]
+    [Serializable]
     public class ProjectData : ScriptableObject
     {
-        private static ProjectData current;
-        public static ProjectData Current => current;
-        
-        [DllImport("__Internal")] private static extern void DownloadFromIndexedDB(string filename, string callbackObjectName, string callbackMethodName);
-        [DllImport("__Internal")] private static extern void SyncFilesToIndexedDB(string callbackObjectName, string callbackMethodName);
-        
+        [JsonIgnore] private static ProjectData current;
+        [JsonIgnore] public static ProjectData Current => current;
```

Comment:
It may be interesting to split this up in a Repository class and Domain object. The repository can do the loading and saving, and if it also manages which project is the current project, it can use events to inform other parts of the system (such as the LayerUIManager) that the project has changed and they need to act accordingly.

It would also make the code more independent / adhere to the Single Responsbility principle; making it more resistant to change

Reply:
Also: ProjectData is a ScriptableObject; and thus a singleton of itself. Why is there a current method if you can attach it to any monobehaviour?

Reply:
agreed with your first comment, but I would like this to be done in a separate PR.
the Current is used in non-monobehaviour classes (such as the LayerData to reference the RootLayer)